### ISSUE TITLE
Weird issue(s) with fetch from endpoint and multipart file content

### ISSUE THREAD

**Issue by mvanhalen** [sveltejs/kit#5490]
### Describe the bug

After NPM update have some issues with uploading multipart form content including a file. This used to work fine.

Now get MultipartParser.end(): stream ended unexpectedly when trying the same code in a js endpoint.

Also after this error repeated actions seem not to forward the multipart form data

### Reproduction

Just do form submit with multipart content to a js endpoint.

Try to forward that to another server like [https://kit.svelte.dev/docs/routing#endpoints-body-parsing](https://kit.svelte.dev/docs/routing#endpoints-body-parsing) mentioned here.

`
 export async function post({ url,request, event,locals}) {
    const data= await request.formData();

    const nftsReq = await fetch( 'someurl', {
        method: 'POST',
        body: data
    })
`

Happens with each first upload. The forwarded request seems without data. Seems like the fetch behavior changed?

### Logs

```shell
MultipartParser.end(): stream ended unexpectedly
Error: MultipartParser.end(): stream ended unexpectedly
    at MultipartParser.end (file:///C:/Users/mvanh/source/repos/DeSo/nftz/nftz-front/node_modules/@sveltejs/kit/dist/chunks/multipart-parser.js:326:10)
    at toFormData (file:///C:/Users/mvanh/source/repos/DeSo/nftz/nftz-front/node_modules/@sveltejs/kit/dist/chunks/multipart-parser.js:440:9)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async post (C:\Users\mvanh\source\repos\DeSo\nftz\nftz-front\src\routes\api\account\uploadimage.js:4:17)
    at async render_endpoint (file:///C:/Users/mvanh/source/repos/DeSo/nftz/nftz-front/.svelte-kit/runtime/server/index.js:269:19)
    at async resolve (file:///C:/Users/mvanh/source/repos/DeSo/nftz/nftz-front/.svelte-kit/runtime/server/index.js:3345:11)
    at async Object.handle (/C:\Users\mvanh\source\repos\DeSo\nftz\nftz-front\src\hooks:72:12)
    at async respond (file:///C:/Users/mvanh/source/repos/DeSo/nftz/nftz-front/.svelte-kit/runtime/server/index.js:3290:20)
    at async file:///C:/Users/mvanh/source/repos/DeSo/nftz/nftz-front/node_modules/@sveltejs/kit/dist/vite.js:2324:22
```

### System Info

```shell
Windows, Vercel, latest npm update version
```

### Severity

serious, but I can work around it

**Comment by huyffs** [comment]
This happens to json body too - large post body is cut off prematurely, less than 60kb files works fine.

**Comment by rburgst** [comment]
for me this happens without forwarding the request, just post a large multipart (including 2 files)

**Comment by mvanhalen** [comment]
Yeah @rburgst @huyffs issue seems when requesting the data indeed. Did not try latest updates of today yet.

**Comment by rburgst** [comment]
I can confirm that `1.0.0-next.369` works, `1.0.0-next.370` is broken

**Comment by rburgst** [comment]
note that I am using a plain old `<input type="file">` upload with a normal endpoint.

**Comment by rburgst** [comment]
I am fairly certain that this broke it: https://github.com/sveltejs/kit/pull/5442

**Comment by rburgst** [comment]
I am trying to upload a 126 KB image.

**Comment by mvanhalen** [comment]
> I am fairly certain that this broke it: #5442

That seems very likely

**Comment by Lootwig** [comment]
Does this also only occur on first submit, but works just fine on subsequent submits for anyone else? 1.0.0-next.377 here.

**Comment by tarngerine** [comment]
i'm getting this intermittently as well, super hard to repro, any workarounds? on next.372 here

**Comment by malalecherocks** [comment]
Same here... and when it does not happen some how, supabase is telling me that the file im trying to upload is not a Blob instance... I don't know if that's related, but I didn't touch anything of that, I upgraded and these erros appear...

**Comment by alexmccartneymoore** [comment]
Getting this too when fetching large json body.

**Comment by alexmccartneymoore** [comment]
Not sure if this is relevant to you but this sveltekit change uncovered an issue with my server configuration. My server was proxying using http1.0 not 1.1 (to allow chunked responses).

**Comment by x-GF-x** [comment]
I am still having this issue. 1.0.0-next.391. Trying to post a PDF and the formData being sent in the body is coming through empty. Was working before switch to POST

**Comment by x-GF-x** [comment]
> Perhaps you are sending data without a `content-type` header. Can you provide a repo with reproduction?

Unfortunately I do not know how to mock the post call, unless you have any suggestions? I cannot expose the API I am using. The content-type header is set. I think I'm wrong about no body being passed, it's coming up in my Chrome payload correctly. As far as I can tell, everything is being sent correctly but the API is not finding the keys it is expecting in the body.

**Comment by irishburlybear** [comment]
Same for me also, I am using Directus and trying to upload a file. But I get a 415 Unsupported Media Type.
Used to work before.

**Comment by irishburlybear** [comment]
export async function POST({request}) {
   
    console.log('From Client headers:',request.headers)
    
    const reader = await request.body.getReader()
    let buff

    for (;;){
        const {done, value} = await reader.read()
        if(done) break
        buff += value
        await new Promise((r) => setTimeout(r, 10))
    }

    //console.log(buff.toString('utf-8'))
    

    const req = await fetch(`http://localhost:8055/files`,{
        method:'POST',
        headers: {
            'Content-Type': request.headers.get('Content-Type')
        },

        body: buff
    })
    
    console.log(req.status)
  
    return req
}

**Comment by irishburlybear** [comment]
> > Perhaps you are sending data without a `content-type` header. Can you provide a repo with reproduction?
> 
> Unfortunately I do not know how to mock the post call, unless you have any suggestions? I cannot expose the API I am using. The content-type header is set. I think I'm wrong about no body being passed, it's coming up in my Chrome payload correctly. As far as I can tell, everything is being sent correctly but the API is not finding the keys it is expecting in the body.

`export async function POST({request}) {
   
    console.log('From Client headers:',request.headers)
    
    const reader = await request.body.getReader()
    let buff

    for (;;){
        const {done, value} = await reader.read()
        if(done) break
        buff += value
        await new Promise((r) => setTimeout(r, 10))
    }

    //console.log(buff.toString('utf-8'))
    

    const req = await fetch(`http://localhost:8055/files`,{
        method:'POST',
        headers: {
            'Content-Type': request.headers.get('Content-Type')
        },

        body: buff
    })
    
    console.log(req.status)
  
    return req
}`

**Comment by deadenddeveloper** [comment]
The same problem. Stuck for a couple of hours, no idea how to fix it.
Kit ver: `1.0.0-next.370`

What I've seen at this moment:

**The POST body is limited to 65536 bytes.**

Firstly I tried to send some form to my endpoint, and it was good, even with files attached. I used JSON and sent files as a Base64 string. Then I've noticed that absolutely valid JSON was parsed on the endpoint wrong.

Then I rewrote my code to use `multipart/form-data` and the error was following:

```
MultipartParser.end(): stream ended unexpectedly
```

One more weird thing is that everyhing is ok with body size at the `handle` hook.
